packages/headless/src/features/facets/range-facets/date-facet-set/date-facet-set-slice.ts

```diff
@@ -41,6 +42,11 @@ export const dateFacetSetReducer = createReducer(
       .addCase(deselectAllDateFacetValues, (state, action) => {
         handleRangeFacetDeselectAll<DateFacetRequest>(state, action.payload);
       })
+      .addCase(clearAllFacetValues, (state, _) => {
+        Object.keys(state).forEach((facetId) => {
```

Comment:
Small nitpick, but you can create a new FacetReducerHelper & a new RangeFacetReducerHelper instead or rewriting the loop.